examples/subplots_axes_and_figures/secondary_axis.py

```diff
@@ -0,0 +1,105 @@
+"""
+==============
+Secondary Axis
+==============
+
+Sometimes we want as secondary axis on a plot, for instance to convert
+radians to degrees on the same plot.  We can do this by making a child
+axes with only one axis visible via `.Axes.axes.secondary_xaxis` and
+`.Axes.axes.secondary_yaxis`.
+
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.transforms import Transform
+from matplotlib.ticker import (
+    AutoLocator, AutoMinorLocator)
+
+fig, ax = plt.subplots(constrained_layout=True)
+x = np.arange(0, 360, 1)
+y = np.sin(2 * x * np.pi / 180)
+ax.plot(x, y)
+ax.set_xlabel('angle [degrees]')
+ax.set_ylabel('signal')
+ax.set_title('Sine wave')
+
+secax = ax.secondary_xaxis('top', conversion=[np.pi / 180])
+secax.set_xlabel('angle [rad]')
+plt.show()
+
+###########################################################################
+# The conversion can be a linear slope and an offset as a 2-tuple.  It can
+# also be more complicated, so we allow the user to pass a pair of
+# functions as the conversion interface, the first is the forward conversion,
+# and the second is the inverse.
+#
+# .. note ::
+#
+#   In this case, the xscale of the parent is logarithmic, so the child is
+#   made logarithmic as well.
+
+fig, ax = plt.subplots(constrained_layout=True)
+x = np.arange(0.02, 1, 0.02)
+np.random.seed(19680801)
+y = np.random.randn(len(x)) ** 2
+ax.loglog(x, y)
+ax.set_xlabel('f [Hz]')
+ax.set_ylabel('PSD')
+ax.set_title('Random spectrum')
+
+
+def forward(x):
+    return 1 / x
+
+
+def inverse(x):
+    return 1 / x
```

Comment:
It might be a bit confusing that the example uses the one and only case where forward and inverse are actually the same, although it probably makes sense to you `a/x` to show how a log scale behaves. Would something like `2./np.sqrt(x)` also work here?